chain/txValidator.go

```diff
@@ -271,6 +292,27 @@ func VerifyTransactionWithLedger(txn *transaction.Transaction) error {
 		if len(id) != 0 {
 			return errors.New("ID has be registered")
 		}
+	case pb.NanoPayType:
+		pld := payload.(*pb.NanoPay)
+
+		channelBalance, limit := DefaultLedger.Store.GetNanoPay(
+			BytesToUint160(pld.Sender),
+			BytesToUint160(pld.Recipient),
+			pld.Nonce,
+		)
+
```

Comment:
We probably need to check and return error if nanopay does not exist (channelBalance == limit == 0)

Reply:
Why? This could be first claim, so channelBalance + limit would be 0

Reply:
Because `if limit < header.UnsignedHeader.Height` below will always be true for the first claim. You are right about the first claim, but still we need another condition below to make sure the first claim can go through.

Reply:
done